gpii/node_modules/matchMakerFramework/src/MatchMakerFramework.js

```diff
@@ -81,6 +80,158 @@
             event.fire(matchData, matchMakerPayload);
         });
     };
+	
+    /*
+	 * Returns a string indicating the matchmaker to select from configuration. Depends on the matchmaker payload.
+     */
+    gpii.matchMakerFramework.selectMatchMaker = function (matchMakerPayload) {
+		var result = gpii.matchMakerFramework.GetPreferredMatchMaker(matchMakerPayload);
+		return (result) ? result : gpii.matchMakerFramework.GetCapableMatchMaker(matchMakerPayload);
```

Comment:
Also - both for GetPreferredMatchMaker and GetCapableMatchMaker - this should be filtered by the available matchmakers (ie. the ones from the configuration file -- you can get the list via: `that.options.matchMakers`) - currently several (most) of the unit and integration tests are failing because the system finds the statistical MM to be most suited, but it cannot find a directive for it in the configuration